shared/js/background.js

```diff
@@ -224,11 +223,28 @@ chrome.webRequest.onBeforeRequest.addListener(
             return {redirectUrl: thisTab.downgradeHttpsUpgradeRequest(requestData)}
         }
 
-        // Fetch upgrade rule from db
-        return new Promise ((resolve) => {
-            const isMainFrame = requestData.type === 'main_frame' ? true : false
+        // Make sure https module is ready, check synchronous .isReady property
+        // since we aren't sure if we're in Chrome or Firefox at this point
+        if (!https.isReady) return
+
+        // Is this request from the tab's main frame?
+        const isMainFrame = requestData.type === 'main_frame' ? true : false
+
+        // Fetch upgrade rule from db (synchronous -- Chrome only)
+        if (utils.isChromeBrowser()) {
```

Comment:
This doesn't change. What about setting it as a background variable to avoid calling this so often?

Reply:
Sounds reasonable.